src/main/java/com/questhelper/achievementdiaries/westernprovinces/WesternMedium.java

```diff
@@ -0,0 +1,436 @@
+		tinderbox = new ItemRequirement("Tinderbox", ItemID.TINDERBOX).showConditioned(notApeTeak);
+		rope = new ItemRequirement("Rope", ItemID.ROPE).showConditioned(notEagleFeldip);
+		gnomebowl = new ItemRequirement("Gnomebowl mould", ItemID.GNOMEBOWL_MOULD).showConditioned(notChocolateBomb);
+		gnomebowl.setTooltip("can be purchased at Grand Tree Groceries");
+		gianneDough = new ItemRequirement("Gianne dough", ItemID.GIANNE_DOUGH).showConditioned(notChocolateBomb);
+		gianneDough.setTooltip("can be purchased at Grand Tree Groceries");
+		chocolateBar = new ItemRequirement("Chocolate bar", ItemID.CHOCOLATE_BAR, 4).showConditioned(notChocolateBomb);
+		chocolateBar.setTooltip("can be purchased at Grand Tree Groceries");
+		equaLeaf = new ItemRequirement("Equa leaves", ItemID.EQUA_LEAVES).showConditioned(notChocolateBomb);
+		equaLeaf.setTooltip("can be purchased at Grand Tree Groceries");
+		potOfCream = new ItemRequirement("Pot of cream", ItemID.POT_OF_CREAM, 2).showConditioned(notChocolateBomb);
+		potOfCream.setTooltip("can be purchased at Grand Tree Groceries");
+		chocolateDust = new ItemRequirement("Chocolate dust", ItemID.CHOCOLATE_DUST).showConditioned(notChocolateBomb);
+		chocolateDust.setTooltip("can be purchased at Grand Tree Groceries");
+		crystalSawSeed = new ItemRequirement("Crystal saw seed", ItemID.CRYSTAL_SAW_SEED).showConditioned(notCrystalSaw);
+		pickaxe = new ItemRequirement("Any pickaxe", ItemCollections.getPickaxes()).showConditioned(notMineGold);
+		ogreBellows = new ItemRequirement("Ogre bellows", ItemCollections.getOgreBellows()).showConditioned(notChompyHat);
+		ogreBow = new ItemRequirement("Ogre bow", ItemCollections.getOgreBow()).showConditioned(notChompyHat);
+		ogreArrows = new ItemRequirement("Ogre / brutal arrows", ItemCollections.getOgreBrutalArrows()).showConditioned(notChompyHat);
+		teakLogs = new ItemRequirement("Teak logs", ItemID.TEAK_LOGS);
+
+		combatGear = new ItemRequirement("Combat gear", -1, -1);
+		combatGear.setDisplayItemId(BankSlotIcons.getCombatGear());
+
+
+		food = new ItemRequirement("Food", ItemCollections.getGoodEatingFood(), -1);
+		fairyAccess = new ItemRequirement("Dramen or Lunar staff", ItemCollections.getFairyStaff());
+		seedPod = new ItemRequirement("Royal seed pod", ItemID.ROYAL_SEED_POD);
+
+		inBrimstailCave = new ZoneRequirement(brimstailCave);
+		inPest = new ZoneRequirement(pest);
+		inApeAtoll = new ZoneRequirement(apeAtoll);
+		inStrongholdBasement = new ZoneRequirement(strongholdBasement);
+		inStrongholdFirst = new ZoneRequirement(strongholdFirst);
+		inEagleArea = new ZoneRequirement(eagleArea);
+
+		choppedLogs = new ChatMessageRequirement(
+			inApeAtoll,
+			"<col=0040ff>Achievement Diary Stage Task - Current stage: 1.</col>"
+		);
+		((ChatMessageRequirement) choppedLogs).setInvalidateRequirement(
+			new ChatMessageRequirement(
+				new Conditions(LogicType.NOR, inApeAtoll),
+				"<col=0040ff>Achievement Diary Stage Task - Current stage: 1.</col>"
```

Comment:
Would you happen to know if completing a non-staged task also invalidates an existing step?

Reply:
Yeah it does, anything that is staged gets reseted by completing any other task. It's actually how I tested a few of these on leagues.